TestON/drivers/common/cli/emulator/mininetclidriver.py

```diff
@@ -367,6 +367,64 @@ def fpingHost( self, **pingParams ):
         main.log.info( self.name + ": \n---\n" + response )
         return main.FALSE
 
+    def pingallHosts( self, hostList, pingType='ipv4' ):
+        """
+            Ping all specified hosts with a specific ping type 
+            
+            Acceptable pingTypes: 
+                - 'ipv4' 
+                - 'ipv6'
+        
+            Acceptable hostList:
+                - ['h1','h2','h3','h4']
+                
+            Returns main.TRUE if all hosts specified can reach 
+            each other
+            
+            Returns main.FALSE if one or more of hosts specified
+            cannot reach each other"""
+            
+        if pingType == "ipv4":
+            cmd = " ping -c 1 -i 1 -W 8 " 
+        elif pingType == "ipv6":
+            cmd = " ping6 -c 1 -i 1 -W 8 "
+        else:
+            main.log.warn( "Invalid pingType specified" )
+            return
+
+        try:
+            main.log.info( "Testing reachability between specified hosts" )
+           
+            isReachable = main.TRUE
+
+            for host in hostList:
```

Comment:
Check to see if using something like below reduces one extra for loop and make it more elegant.
hostCombos = list( itertools.combinations( hostList, 2 ) )
for i in xrange( 0, len( hostCombos )):
    pingCmd = hostCombos[i][0] + cmd + hostCombos[i][1]]